parser_test.go

```diff
@@ -2040,3 +2060,32 @@ func TestContainsUnescaped(t *testing.T) {
 		}
 	}
 }
+
```

Comment:
Please add tests for your other two functions `validNetwork` and `portsValid` as well.

Reply:
done